src/components/Select.tsx

```diff
@@ -1,28 +1,31 @@
 import './styles/select.less'
 
 import React from 'react'
-import PropTypes from 'prop-types'
 import classNames from 'classnames'
 
-/**
- * @typedef {Object} Option
- * @property {string | number | undefined} [key]
- * @property {string | number} [value]
- * @property {string | number | undefined} [text]
- * @property {boolean | undefined} [disabled]
- */
+interface Option {
+    /** Optional key for each option. If not provided `value` is used. */
+    key?: string | number
+    /** Value of the option. */
+    value: string | number
+    /** Text to display for the option. */
+    text?: string | number
+    /** Whether the options is disabled or not. */
+    disabled?: boolean
+}
 
-/**
- * @typedef {Object} Props
- * @property {string | undefined} [className]
- * @property {boolean | undefined} [disabled]
- * @property {string | number} value
- * @property {(value: Props['value']) => void} onChange
- * @property {Option[] | undefined} [options]
- */
+interface Props {
+    className?: string
+    disabled?: boolean
+    /** Currently selected value. */
+    value: string | number
+    /** Callback for the change event. Will be called with the next value (not the full event). */
+    onChange?: (value: Props['value']) => void
+    /** Options that are rendered in the select. */
+    options?: Option[]
```

Comment:
Does it ever make sense to have Select without options? It seems to me that having optional Array is often bad style, and empty default array would be better.

I'm not sure about all consequences of this, but would `options = []` default work?

Reply:
Ref.: https://twist.com/a/1585/ch/190200/t/1504732/c/58090586

Reply:
Yeah, that makes sense. Makes the typing simpler to read and understand. Done!